### ISSUE TITLE
Connect Giiker cube under Linux/Ubuntu

### ISSUE THREAD

**Issue by peisenh** [cs0x7f/cstimer#109]
I have some strange behaviour here:

If a try to use cstimer under Linux/Chome and connect my Giiker i3se the cube is not shown in the windows which is searching for bluetooth devices - so I cannot connect.

Using chrome://bluetooth-internals/#devices a can see the device. If a then inspect and connect the device with bluetooth-internals: Trying then to use cstimer shows the device, I can connect and everything is working fine. Trying without this connect on bluetooth will not show the cube.

This is strange since e.g. https://avaver.github.io can be used without this workaround using the same system/browser.

Anybody else having this problem? Any idea which could be the cause of the problem?

Greetings,
Peter

**Comment by cs0x7f** [comment]
I cannot reproduce the bug.
The connection logic of cstimer and avaver.github.io is almost the same. What's the device name of your cube?

**Comment by peisenh** [comment]
I did some further tests.

Looking into the source there seems to be a filter of services for the bluetooth connection in bluetooth.js
(that is the difference since e.g. avaver.github.io is only filtering namePrefix)

One of he filtered service is showing up for the cube:
        UUID: Generic Access Profile    (00001800-0000-1000-8000-00805f9b34fb)
        UUID: Generic Attribute Profile (00001801-0000-1000-8000-00805f9b34fb)
        UUID: Device Information        (0000180a-0000-1000-8000-00805f9b34fb)
        UUID: Unknown                   (0000aaaa-0000-1000-8000-00805f9b34fb)
        UUID: Unknown                   (0000aadb-0000-1000-8000-00805f9b34fb)

So the cube should be shown in the bluetooth windows but is missing.

After deleting the services filters but leaving the namePrefix filters from bluetooth.js the cube is then shown and everything is fine.

Looks like some weird behaviour/bug of service filtering under Linux and Chrome.

Did you really try to reproduce this under linux or some other system?

Greetings,
Peter

**Comment by cs0x7f** [comment]
I tried chromium in ubuntu 18.04, everything works as expected for my Giiker i3S.

**Comment by peisenh** [comment]
Tested here with Ubuntu 18.04. Both chromium and latest stable chrome from google showing the same. Also tested with Ubuntu 18.04 and chrome stable 14 days ago under different hardware.

Cube I am using is Giiker i3se (2019 version), so this is a difference.

**Comment by sonofhypnos** [comment]
I am not sure whether this belongs here, but if it does the title of this issue should probably be changed
I have the exact same problem with my Giiker i3se on my phone (android/chrome):

I was able to connect the cube with [chrome://bluetooth-internals/#devices](url) and only after that I was able to connect the cube with on [https://cstimer.net](url) (although it is a bit inconsistent and sometimes it still does not work).
On https://avaver.github.io the cube connected without any problem. 

So my guess would be that this has something to do with the cube and not the operating system.
